### PR TITLE
Cask: add dictionary artifact

### PR DESCRIPTION
**Pull request by Eitot**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren’t other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [X] Have you added an explanation of what your changes do and why you’d like us to include them?
- [X] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

This adds a new artifact type for dictionaries. Presently there are only two dictionaries at Caskroom that would use this, thus this PR is mostly for sake of completion and consistency. That said, the “service” and “input_method” artifacts are used only a couple of times as well.

### CODE REVIEW

File: manpages/brew-cask.1 [hunk 1]
```diff
@@ -162,6 +162,10 @@ Target location for Preference Panes\. The default value is \fB~/Library/Prefere
 Target location for QuickLook Plugins\. The default value is \fB~/Library/QuickLook\fR\.
 .
 .TP
+\fB\-\-dictionarydir=<path>\fR
```
Comment (jawshooah) [inline manpages/brew-cask.1]: Don't edit this file directly. Make the change in [brew-cask.1.md](https://github.com/Homebrew/brew/blob/8995d8b9e5ccaf3c56aceff655c51f924737a11e/Library/Homebrew/manpages/brew-cask.1.md), then run `brew man` to generate the manpage.

Comment (Eitot) [inline manpages/brew-cask.1]: Done.

Comment (jawshooah) [inline manpages/brew-cask.1]: Sorry if I wasn't clear; you still need to commit `brew-cask.1.md` and `brew-cask.1`.

**Comment by jawshooah** [comment]
This also needs to be rebased now, as there are merge conflicts.

File: manpages/brew-cask.1 (hunk 1 above)
Comment (Eitot) [inline manpages/brew-cask.1]: No, that was me being dense. 😅

**Pull request merged**